### ISSUE TITLE
RTK Query | toInvalidate.values() is not a function

### ISSUE THREAD

**Issue by Aliheym** [reduxjs/redux-toolkit#2473]
I prepared small example of this error:

https://codesandbox.io/s/focused-kilby-s4930g

As you can see, I use the basic RTK functionality - tag mutation and invalidation.

Everything works correctly in the latest versions of Chrome, Firefox and other browsers.

![image](https://user-images.githubusercontent.com/72506898/176660481-0e46f292-3209-46d4-80eb-91c92366aaa3.png)

When I try to open this example in older browsers like Safari 10 or Chrome 65, I get an error. 
![image](https://user-images.githubusercontent.com/72506898/176660530-c4ee0a1d-ff14-4ef8-a6d6-b9a22dd5d891.png)

Is it because it doesn't support those browsers?

**Comment by markerikson** [comment]
That seems very odd.  In the relevant line of code, `toInvalidate` is a `Set`, and according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set it's been supported in Chrome since v38.

**Comment by Aliheym** [comment]
@markerikson As I see `Set.prototype.values` is working in Chrome v65:

![image](https://user-images.githubusercontent.com/72506898/176738905-f352a8a6-5512-43ad-af46-bf6bb65525cf.png)

**Comment by phryneas** [comment]
The code in question here is

```js
  function selectInvalidatedBy(state, tags) {
    var _a;
    var apiState = state[reducerPath];
    var toInvalidate = new Set();
    for (var _i = 0, _e = tags.map(expandTagDescription); _i < _e.length; _i++) {
      var tag = _e[_i];
      var provided = apiState.provided[tag.type];
      if (! provided) {
        continue;
      }
      var invalidateSubscriptions = (_a = tag.id !== void  0 ? provided[tag.id] : flatten(Object.values(provided))) != null ? _a : [];
      for (var _f = 0, invalidateSubscriptions_1 = invalidateSubscriptions; _f < invalidateSubscriptions_1.length; _f++) {
        var invalidate = invalidateSubscriptions_1[_f];
        toInvalidate.add(invalidate);
      }
    }
    return flatten(Array.from(toInvalidate.values()).map(function (queryCacheKey) {
      var querySubState = apiState.queries[queryCacheKey];
      return querySubState ? [{
        queryCacheKey: queryCacheKey,
        endpointName: querySubState.endpointName,
        originalArgs: querySubState.originalArgs
      }] : [];
    }));
  }
```

So this should be a normal `Set.prototype.values` call - and that should be supported since Safari 8 and Chrome 38 - the only thing I can think of is that you have some kind of broken `Set` polyfill in place.

**Comment by phryneas** [comment]
Yeah with something basic like that, if you are targeting those outdated browsers, you should really add some kind of polyfill like https://github.com/zloirock/core-js to your app.

**Comment by Aliheym** [comment]
Yes, got you. Thanks.

**Comment by phryneas** [comment]
To go deeper into what I said earlier about "these browsers should not be connected to the internet", please see [this on chrome releases](https://en.wikipedia.org/wiki/Google_Chrome_version_history). The "extended stable channel" of chrome contains the oldest chrome versions that still get security updates. The oldest version of chrome supported by Google is Chrome 102.

Surfing the internet with an older version has a very high risk of compromising your system and getting you hacked or infected with viruses.

Similarly, Safari 10 has seen the last security update in 2017 - it is very safe to assume that there are dozens of bugs in that old unsupported version that puts every system running that browser in the internet at a serious risk.
